pyirf/interpolation/visible_edges_extrapolator.py

```diff
@@ -0,0 +1,215 @@
+        Univ. of Wisconsin-Madison. https://apps.dtic.mil/sti/pdfs/ADA144660.pdf
+        """
+
+        super().__init__(grid_points, params)
+
+        # Test wether m is a number
+        try:
+            m > 0
+        except TypeError:
+            raise ValueError(f"Only positiv integers allowed for m, got {m}.")
+
+        # Test wether m is a finite, positive integer
+        if (m <= 0) or ~np.isfinite(m) or (m != int(m)):
+            raise ValueError(f"Only positiv integers allowed for m, got {m}.")
+
+        self.m = m
+
+    def extrapolate(self, target_point):
+        """
+        Takes a grid of scalar values for a bunch of different parameters
+        and extrapolates it to given value of those parameters. Falls back
+        to ParametrizedNearestSimplexExtrapolator's behavior for 1D grids or
+        cases, where only one simplex is visible in 2D.
+
+        Parameters
+        ----------
+        target_point: numpy.ndarray
+            Value for which the extrapolation is performed (target point)
+
+        Returns
+        -------
+        values: numpy.ndarray, shape=(1,...)
+            Extrapolated values
+
+        """
+        if self.grid_dim == 1:
+            extrapolant = super().extrapolate(target_point)
+        elif self.grid_dim == 2:
+            visible_facet_points = find_visible_facets(self.grid_points, target_point)
+
+            if visible_facet_points.shape[0] == 1:
+                extrapolant = super().extrapolate(target_point)
+            else:
+                simplices_points = self.triangulation.points[
+                    self.triangulation.simplices
+                ]
+
+                visible_simplices_indices = np.array(
+                    [
+                        find_simplex_to_facet(simplices_points, facet)
+                        for facet in visible_facet_points
+                    ]
+                )
+
+                extrapolation_weigths = compute_extrapolation_weights(
+                    visible_facet_points, target_point, self.m
+                )
+
+                extrapolation_weigths = extrapolation_weigths.reshape(
+                    extrapolation_weigths.shape[0],
+                    *np.ones(self.params.ndim - 1, "int"),
+                )
+
+                # Function has to be copied outside list comprehention as the super() short-form
+                # cannot be used inside it (at least until python 3.11)
+                extrapolate2D = super()._extrapolate2D
+
+                simplex_extrapolations = np.array(
+                    [
+                        extrapolate2D(
+                            self.triangulation.simplices[ind], target_point
+                        ).squeeze()
+                        for ind in visible_simplices_indices
+                    ]
+                )
+
+                extrapolant = np.sum(
+                    extrapolation_weigths * simplex_extrapolations, axis=0
+                )[np.newaxis, :]
```

Comment:
Missing else / nice error for n > 2 ?

Reply:
The n>2 case is handled by `super().__init__()`, assuring that a >2D grid cannot be used to create the Extrapolator instance.